plugins/CoreHome/Controller.php

```diff
@@ -185,10 +187,44 @@ protected function setDateTodayIfWebsiteCreatedToday()
     public function index()
     {
         $this->setDateTodayIfWebsiteCreatedToday();
+
+        if ($this->noRecordedData()) {
+            return $this->showSiteNoData();
+        }
+
         $view = $this->getDefaultIndexView();
         return $view->render();
     }
 
+    private function noRecordedData()
+    {
+        $sql = "SELECT COUNT(*)
+                FROM " . Common::prefixTable('log_visit') . "
+                WHERE idsite=?";
+
+        $count = Db::fetchOne($sql, array($this->idSite));
+
+        return $count == 0;
+    }
```

Comment:
it should be possible to do something like `SELECT idvisit FROM piwik_log_visit WHERE idsite=? limit 1` then it should have an effect.